go/private/mode.bzl

```diff
@@ -122,6 +122,8 @@ def get_mode(ctx, go_toolchain, cgo_context_info, go_config_info):
         tags.append("race")
     if msan:
         tags.append("msan")
+    if pure:
+        tags.append("purego")
```

Comment:
Looking at the CI, this doesn't seem to be the correct place to add the tag 😞

Reply:
Something weird is going on in the Google protobuf go package. This PR looks fine, maybe that package isn't following the convention correctly?

Reply:
You'd expect it to include [reflect/protoreflect/value_pure.go](https://github.com/protocolbuffers/protobuf-go/blob/3ebf7dd8a5331895cd3ffd849d6ebd313c509e0f/reflect/protoreflect/value_pure.go#L6). I'll have a look into it.

Reply:
The `gazelle` generation does not include the `value_pure.go`, `reflect/protoreflect/BUILD.bazel` library looks like so:

```
go_library(
    name = "protoreflect",
    srcs = [
        "methods.go",
        "proto.go",
        "source.go",
        "source_gen.go",
        "type.go",
        "value.go",
        "value_equal.go",
        "value_union.go",
        "value_unsafe_go120.go",
        "value_unsafe_go121.go",
    ],
    importpath = "google.golang.org/protobuf/reflect/protoreflect",
    visibility = ["//visibility:public"],
    deps = [
        "//encoding/protowire",
        "//internal/pragma",
    ],
)
```

Which is missing the `value_pure.go`. Adding it makes it work.

Do we also need a patch for `gazelle` here? I would expect it to include the `value_pure.go` if `--build_tags` is not specified.

Reply:
I'm figuring out why `gazelle` is not adding that file. Once I've worked it out, I'll raise a PR on `gazelle`.

Reply:
@fmeum bazelbuild/bazel-gazelle#1767 ignores the `purego` tag so that the `value_pure.go` gets added to the `srcs`.

I imagine we need to land that, do a `gazelle` release, update the minimum version here and then land this?

Reply:
I've bumped to `gazelle@0.36.0` to include the `purego` patch.

Reply:
The CI still seems to be not adding `value_pure.go` but locally, building that part works:

```
~/g/g/b/rules_go (purego|✔) $ bazelisk query @org_golang_google_protobuf//reflect/protoreflect/...
@org_golang_google_protobuf//reflect/protoreflect:go_default_library
@org_golang_google_protobuf//reflect/protoreflect:protoreflect
@org_golang_google_protobuf//reflect/protoreflect:protoreflect_test
~/g/g/b/rules_go (purego|✔) $ bazelisk build @org_golang_google_protobuf//reflect/protoreflect
INFO: Analyzed target @@org_golang_google_protobuf//reflect/protoreflect:protoreflect (60 packages loaded, 9830 targets configured).
INFO: Found 1 target...
Target @@org_golang_google_protobuf//reflect/protoreflect:protoreflect up-to-date:
  bazel-bin/external/org_golang_google_protobuf/reflect/protoreflect/protoreflect.x
INFO: Elapsed time: 13.779s, Critical Path: 0.85s
INFO: 6 processes: 1 internal, 5 linux-sandbox.
INFO: Build completed successfully, 6 total actions
```

Reply:
🤦 of course there is a world of `WORKSPACE` and I haven't updated that.

Reply:
Needed to update the `third_party/org_golang_google_protobuf-gazelle.patch` due to new files being added from the Gazelle 0.36.0 upgrade.